src/Function.cpp

```diff
@@ -875,6 +875,16 @@ GenerateFunctions(void)
 			ERROR_RETURN();
 		}
 	}
+	if(CGOptions::trans_memory_atomic()){
+		// Total number of transaction block to be generated
+		int atomic_blk_cnt = rnd_upto(FuncListSize()) + 1 ;
```

Comment:
I don't like. I would rather set a small probability that will drive the emission of `__transaction_atomic`. You then don't need to randomly select a function, etc.

Reply:
Done. Please check it.